packages/core/src/modules/discover-features/FeatureRegistry.ts

```diff
@@ -0,0 +1,52 @@
+import type { FeatureQuery, Feature } from './models'
+
+import { injectable } from '../../plugins'
+
+@injectable()
+class FeatureRegistry {
+  private features: Feature[] = []
+
+  /**
+   * Register a single or set of Features on the registry
+   *
+   * @param features set of {Feature} objects
+   */
+  public register(...features: Feature[]) {
+    for (const feature of features) {
+      const index = this.features.findIndex((item) => item.type === feature.type && item.id === feature.id)
+
+      if (index > -1) {
+        this.features[index] = this.features[index].combine(feature)
+      } else {
+        this.features.push(feature)
+      }
+    }
+  }
+
+  /**
+   * Perform a set of queries in the registry, supporting wildcards (*) as
+   * expressed in Aries RFC 0557.
+   *
+   * @see https://github.com/hyperledger/aries-rfcs/blob/560ffd23361f16a01e34ccb7dcc908ec28c5ddb1/features/0557-discover-features-v2/README.md
+   *
+   * @param queries set of {FeatureQuery} objects to query features
+   * @returns array containing all matching features (can be empty)
+   */
+  public query(...queries: FeatureQuery[]) {
+    const output = []
+    for (const query of queries) {
+      const items = this.features.filter((item) => item.type === query.featureType)
+      if (query.match === '*') {
```

Comment:
maybe some inline comments here would help to understand what the different paths mean?